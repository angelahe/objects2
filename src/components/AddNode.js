import React from 'react';
import closebtn from '../images/close_FFFFFF.png';
import "../styles/Styles140.css";

const AddNode = (props) => {

  function handleCloseClick() {
    props.closeClicked("create");
  }

  function handleAddClick() {
//        console.log("in handleAddClick function");
    const subject = document.getElementById("subject").value;
    const amount = Number(document.getElementById("amount").value);
    props.createClicked(subject, amount);
  }

  return(
    <div>
      <div className = "ItemBox AppHeader">
        <span className= "AddItem">Add Subject</span>
        <button className="AppBtn" onClick={handleCloseClick}>
          <img className="btnImg" src={closebtn} alt="Close"/>
        </button>
      </div>
      <div className = "ItemBox">
        <span className = "DetailText">Subject:</span>
        <input className="InputText" id="subject" autoFocus={true}></input><br/>
        <span className = "DetailText">Amount:</span>
        <input className = "InputText" id="amount" type="number"></input><br/><br/>
        <button className="btnText" onClick={handleAddClick}>Create Account</button>

      </div>
    </div>
  )
}
export default AddNode;