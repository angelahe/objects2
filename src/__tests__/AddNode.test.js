import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import AddNode from "../components/AddNode";
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddNode />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<AddNode />);
});

test('Starter renders correctly', () => {
  const tree = renderer.create(<AddNode />).toJSON();
  expect(tree).toMatchSnapshot();
});