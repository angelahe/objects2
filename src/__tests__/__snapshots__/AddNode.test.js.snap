// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starter renders correctly 1`] = `
<div>
  <div
    className="ItemBox AppHeader"
  >
    <span
      className="AddItem"
    >
      Add Subject
    </span>
    <button
      className="AppBtn"
      onClick={[Function]}
    >
      <img
        alt="Close"
        className="btnImg"
        src="close_FFFFFF.png"
      />
    </button>
  </div>
  <div
    className="ItemBox"
  >
    <span
      className="DetailText"
    >
      Subject:
    </span>
    <input
      autoFocus={true}
      className="InputText"
      id="subject"
    />
    <br />
    <span
      className="DetailText"
    >
      Amount:
    </span>
    <input
      className="InputText"
      id="amount"
      type="number"
    />
    <br />
    <br />
    <button
      className="btnText"
      onClick={[Function]}
    >
      Create
    </button>
  </div>
</div>
`;
